### PR TITLE
[14.0][ADD]  sale_promotion_discount_in_field_order_line_link: new module

### PR DESCRIPTION
**Pull request by HekkiMelody**
New bridge module

First two commits contain minor tweaks to `sale_promotion_discount_in_field` and `sale_coupon_domain_product_discount_in_field` meant to facilitate the new module.

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

File: sale_coupon_domain_product_discount_in_field/models/sale_order.py [hunk 1]
```diff
@@ -13,8 +13,8 @@ def _set_reward_discount_for_lines(self, program):
             and program.discount_apply_on_domain_product
         ):
             domain = program.rule_products_domain
-            super()._set_reward_discount_for_lines(
+            return super()._set_reward_discount_for_lines(
                 program.with_context(promo_domain_product=domain)
             )
         else:
-            super()._set_reward_discount_for_lines(program)
+            return super()._set_reward_discount_for_lines(program)
```
Comment (PicchiSeba) [inline sale_coupon_domain_product_discount_in_field/models/sale_order.py]: **nitpick**: we could return just once at the end of the function and save the program in its own variable.
I think done this way, our objective is more readable.

```suggestion
             program = program.with_context(promo_domain_product=domain)
    return super()._set_reward_discount_for_lines(program)
```

**Review by PicchiSeba — COMMENTED** [review]
Code review: readability improvement

File: sale_coupon_domain_product_discount_in_field/models/sale_order.py (hunk 1 above)
Comment (HekkiMelody) [reply sale_coupon_domain_product_discount_in_field/models/sale_order.py]: done

**Review by PicchiSeba — APPROVED** [review]
Code review: LGTM

**Comment by francesco-ooops** [comment]
@pedrobaeza merge this one too? thanks!

File: sale_promotion_discount_in_field/tests/test_promotion_discount_in_field.py [hunk 2]
```diff
@@ -2,11 +2,9 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
 from odoo.exceptions import UserError
-from odoo.tests import tagged
 from odoo.tests.common import SavepointCase
 
 
-@tagged("post_install", "-at_install")
 class CetmixTestSaleCouponProgram(SavepointCase):
```
Comment (pedrobaeza) [inline sale_promotion_discount_in_field/tests/test_promotion_discount_in_field.py]: You can't put a company name in the test class

**Review by pedrobaeza — DISMISSED** [review]

File: sale_promotion_discount_in_field/tests/test_promotion_discount_in_field.py (hunk 2 above)
Comment (HekkiMelody) [reply sale_promotion_discount_in_field/tests/test_promotion_discount_in_field.py]: didn't want to add to the diff, but you're right. Fixed.

**Comment by francesco-ooops** [comment]
@pedrobaeza good to go now?

**Comment by francesco-ooops** [comment]
@pedrobaeza go? :)

**Pull request merged**